discovery-provider/src/tasks/entity_manager/app_delegate.py

```diff
@@ -128,6 +187,33 @@ def create_app_delegate(params: ManageEntityParameters):
     return delegate_record
 
 
+def delete_app_delegate(params: ManageEntityParameters):
+    metadata = get_delete_app_delegate_metadata_from_raw(params.metadata_cid)
+    if not metadata:
+        raise Exception(
+            "Invalid Revoke AppDelegate Transaction, unable to parse metadata"
+        )
+    validate_app_delegate_tx(params, metadata)
+    address = metadata["address"]
+    existing_delegate = params.existing_records[EntityType.APP_DELEGATE][address]
+    if metadata["address"] in params.new_records[EntityType.APP_DELEGATE]:
+        existing_delegate = params.new_records[EntityType.APP_DELEGATE][address][-1]
+
+    revoked_delegate = copy_record(
+        existing_delegate,
+        params.block_number,
+        params.event_blockhash,
+        params.txhash,
+        params.block_datetime,
+    )
+
+    revoked_delegate.is_revoked = True
```

Comment:
Should we make the `revoked`/`deleted` terminology consistent? Cause this is essentially just doing a "delete" right?

Reply:
maybe revoke for the 1 way deletion

Reply:
yeah agreed! will keep it delete for app delegate and revoked for delegation (since it can be reactivated)